build.gradle

```diff
@@ -88,10 +88,24 @@ ospackage {
 buildRpm {
     arch = 'NOARCH'
     addParentDirs = false
-    archiveName "${packageName}-${version}.rpm"
+    finalizedBy 'renameRpm'
+    task renameRpm(type: Copy) {
+        from("$buildDir/distributions")
```

Comment:
Is it possible to reuse this task for both deb and rpm ?

Reply:
This is not a custom task class. It doesn't take parameters like methods. I don't see a way to reuse the task.

Reply:
You can dynamically create new task as template

Reply:
Something like this 

```
buildRpm {
    arch = 'NOARCH'
    addParentDirs = false
    finalizedBy renameTask(archiveName, packageName, 'rpm')
}

buildDeb {
    arch = 'all'
    finalizedBy renameTask(archiveName, packageName, 'deb')
}

def renameTask(archiveName, packageName, arch) {
    return tasks.create("rename${arch}", Copy) {
        from("$buildDir/distributions")
        into("$buildDir/distributions")
        include archiveName
        rename archiveName, "${packageName}-${version}.${arch}"
        doLast { delete file("$buildDir/distributions/$archiveName") }
    }
}
```

Reply:
@sujithvm I've updated the changes.